leopard.nim

```diff
@@ -0,0 +1,329 @@
+import pkg/stew/results
+import pkg/upraises
+
+import ./leopard/wrapper
+
+export LEO_ALIGN_BYTES, results
+
+push: {.upraises: [].}
+
+const
+  LeopardBadCodeMsg = "Bad RS code"
+  LeopardInconsistentSizeMsg =
+    "Buffer sizes must all be the same multiple of 64 bytes"
+  LeopardNeedLessDataMsg = "Too much recovery data received"
+  LeopardNotEnoughDataMsg = "Buffer counts are too low"
+
+  MinBufferSize* = 64.uint
+  MinSymbols* = 1.uint
+  MaxTotalSymbols* = 65536.uint
+
+type
+  Data* = seq[seq[byte]]
+
+  LeopardDefect* = object of Defect
+
+  # It should not be necessary to redefine LeopardResult, but if that's not
+  # done here then defining LeopardError as `object of CatchableError` will
+  # cause a mystery crash at compile-time (symbol not found). Can workaround by
+  # defining as just `object`, but then when trying to work with LeopardResult
+  # errors in e.g. tests/test_leopard.nim the same mystery crash happens at
+  # compile-time. The problem may be related to use of importcpp in
+  # leopard/wrapper.nim, so it could be a compiler bug. By redefining
+  # LeopardResult in this module (and casting wrapper.LeopardResult values) the
+  # the problem is avoided.
+  LeopardResult* = enum
+    LeopardNotEnoughData    = -11.cint # Buffer counts are too low
+    LeopardNeedLessData     = -10.cint # Too much recovery data received
+    LeopardInconsistentSize =  -9.cint # Buffer sizes must all be the same multiple of 64 bytes
+    LeopardBadCode          =  -8.cint # Bad RS code
+    LeopardCallInitialize   = wrapper.LeopardCallInitialize
```

Comment:
I'm not entirely sure whats the need for this extension of the base type?

Reply:
The `LeopardResult` enum from the underlying lib makes a convenient `T` for `Result[T, E]` in the higher-level wrapper in `leopard.nim`, but when working on it I discovered that error codes `-7..-1` (i.e. the error messages they correspond too) weren't quite expressive enough. So extending the enum seemed natural and convenient.